pylink/jlink.py

```diff
@@ -3044,11 +3044,19 @@ def register_read(self, register_index):
 
         Args:
           self (JLink): the ``JLink`` instance
-          register_index (int): the register to read
+          register_index (int/str): the register to read
 
         Returns:
           The value stored in the given register.
         """
+        # TODO: rename 'register_index' to 'register'
+        if isinstance(register_index, str):
+            regs = list(self.register_name(idx) for idx in self.register_list())
+            try:
+                register_index = regs.index(register_index)
+            except ValueError:
+                raise errors.JLinkException(
```

Comment:
Hm, I think it might be better to word this as `No register found matching name: <register_name> (available registers: <name1>, <name2>...`